includes/Core/User_Input/User_Input.php

```diff
@@ -0,0 +1,384 @@
+		Context $context,
+		Authentication $authentication = null,
+		Options $options = null,
+		User_Options $user_options = null
+	) {
+		$this->authentication          = $authentication ?: new Authentication( $context );
+		$this->site_specific_questions = new Site_Specific_Questions( $options ?: new Options( $context ) );
+		$this->user_specific_questions = new User_Specific_Questions( $user_options ?: new User_Options( $context ) );
+	}
+
+	/**
+	 * Registers functionality.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function register() {
+		$this->site_specific_questions->register();
+		$this->user_specific_questions->register();
+
+		add_filter(
+			'googlesitekit_rest_routes',
+			function( $routes ) {
+				return array_merge( $routes, $this->get_rest_routes() );
+			}
+		);
+	}
+
+	/**
+	 * Gets related REST routes.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array List of REST_Route objects.
+	 */
+	private function get_rest_routes() {
+		$can_authenticate = function() {
+			return current_user_can( Permissions::AUTHENTICATE );
+		};
+
+		return array(
+			new REST_Route(
+				'core/user/data/user-input-settings',
+				array(
+					array(
+						'methods'             => WP_REST_Server::READABLE,
+						'callback'            => function() {
+							return rest_ensure_response( $this->get_answers() );
+						},
+						'permission_callback' => $can_authenticate,
+					),
+					array(
+						'methods'             => WP_REST_Server::CREATABLE,
+						'callback'            => function( WP_REST_Request $request ) {
+							$data = $request->get_param( 'data' );
+
+							if ( ! isset( $data['settings'] ) || ! is_array( $data['settings'] ) ) {
+								return new WP_Error(
+									'rest_missing_callback_param',
+									__( 'Missing settings data.', 'google-site-kit' ),
+									array( 'status' => 400 )
+								);
+							}
+
+							return rest_ensure_response(
+								$this->set_answers(
+									$data['settings']
+								)
+							);
+						},
+						'permission_callback' => $can_authenticate,
+						'args'                => array(
+							'data' => array(
+								'type'       => 'object',
+								'required'   => true,
+								'properties' => array(
+									'settings' => array(
+										'type'      => 'object',
+										'required'  => true,
+										'questions' => array_fill_keys(
+											array_keys( static::$questions ),
+											array(
+												'type'  => 'array',
+												'items' => array( 'type' => 'string' ),
+											)
+										),
+									),
+								),
+							),
+						),
+					),
+				)
+			),
+		);
+	}
+
+	/**
+	 * Gets user input settings.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array|WP_Error User input settings.
+	 */
+	public function get_answers() {
+		$data = array(
+			'site' => $this->site_specific_questions->get(),
+			'user' => $this->user_specific_questions->get(),
+		);
+
+		$questions = static::$questions;
+		$settings  = array_merge( $data['site'], $data['user'] );
```

Comment:
Can't we use `array_merge` directly as we're not using the `$data` variable after deriving `$settings` from it?

Reply:
Updated, thank you!